homeassistant/components/ozw/light.py

```diff
@@ -253,13 +266,12 @@ def _calculate_rgb_values(self):
             temp_cold = self._white
 
         # Calculate color temps based on white LED status
-        if temp_cold > 0:
-            self._ct = round(TEMP_COLOR_MAX - ((temp_cold / 255) * TEMP_COLOR_DIFF))
-        # Only used if CW channel missing
-        elif temp_warm > 0:
-            self._ct = round(TEMP_COLOR_MAX - temp_warm)
+        if temp_cold or temp_warm:
+            self._ct = round(
+                self._max_mireds
+                - ((temp_cold / 255) * (self._max_mireds - self._min_mireds))
```

Comment:
There's no case for `temp_warm` anymore, right? Is that correct?

Reply:
You do need it for one case, when `temp_warm > 0` and `temp_cold == 0`, aka when at 100% warm.   The previous work had the following lines, which cause the slider to bounce over at `temp_warm == 255`. 

```python
        elif temp_warm > 0:
            self._ct = round(TEMP_COLOR_MAX - temp_warm)
```

Reply:
So it's ok to always take `self._max_mireds` as color_temp for that case? Am I doing the math correctly?

Reply:
Correct.  If temp_warm = 255 (the only time temp_cold would be 0, since temp_warm = max_mireds - temp_cool) it will always be max_mireds.

Reply:
Ok. :+1: